src/date.rs

```diff
@@ -390,6 +390,10 @@ where
 }
 
 impl<Tz: TimeZone> Datelike for Date<Tz> {
+    #[inline]
+    fn leap_year(&self) -> bool {
```

Comment:
Let's call this `in_leap_year()`, and keep this addition in a separate commit.

Reply:
it's already called `leap_year` on the `NaiveDate` type, and there are a bunch of other functions like this in the trait and on `NaiveDate` where the names match. wouldn't it feel a bit weird to break that convention here?

Reply:
I guess it doesn't really matter tho, because if the `num_days` logic is moved back to the `Month` type then this function isn't even needed for this PR anymore.